docs/Toy/toy/dialect.py

```diff
@@ -270,23 +290,27 @@ class ReshapeOp(Operation):
     ```
     """
     name: str = 'toy.reshape'
-    arguments: Annotated[VarOperand, AnyTensorTypeF64]
+    arg: Annotated[Operand, AnyTensorTypeF64]
     # We expect that the reshape operation returns a statically shaped tensor.
     res: Annotated[OpResult, TensorTypeF64]
 
     @classmethod
-    def from_input(cls: type[ReshapeOp], input: SSAValue,
+    def from_input(cls: type[ReshapeOp], arg: SSAValue,
                    shape: List[int]) -> ReshapeOp:
-        assert isinstance(input.typ, TensorType | UnrankedTensorType)
-        element_type = cast(Float64Type,
-                            cast(TensorType[Any], input.typ).element_type)
-        t = AnyTensorType.from_type_and_list(element_type, shape)
-        return cls.create(result_types=[t], operands=[input])
+        assert isa(arg.typ, AnyTensorTypeF64)
```

Comment:
My guess is that those `assert` should be `return ...` instead?

Reply:
Or a ValueException?

Reply:
But that constraint is already on the `arg`, meaning that we'll actually get a VerifyException in practice, and the assert will never be hit

Reply:
Can't wait for typed `SSAValue`s

Reply:
Migrated to `ValueError`